monai/metrics/mmd.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Callable
+
+import torch
+
+from monai.metrics.metric import Metric
+
+
+class MMDMetric(Metric):
+    """
+    Unbiased Maximum Mean Discrepancy (MMD) is a kernel-based method for measuring the similarity between two
+    distributions. It is a non-negative metric where a smaller value indicates a closer match between the two
+    distributions.
+
+    Gretton, A., et al,, 2012.  A kernel two-sample test. The Journal of Machine Learning Research, 13(1), pp.723-773.
+
+    Args:
+        y_transform: Callable to transform the y tensor before computing the metric. It is usually a Gaussian or Laplace
+            filter, but it can be any function that takes a tensor as input and returns a tensor as output such as a
+            feature extractor or an Identity function.
+        y_pred_transform: Callable to transform the y_pred tensor before computing the metric.
```

Comment:
Can we have an example how to write the transform? Also I am wondering if the term `transform` is appropiate as we have `monai.transforms`. Thanks!

Reply:
@mingxin-zheng I agree that the term transform might be ambiguous, especially in torch nomenclature. We could use `y_kernel` or `y_kernel_transform`. 

I used that nomenclature for speed when I implemented it.

Reply:
I've renamed this `y_mapping` to better reflect that we're suppling $\varphi$ rather than the kernel $k$ as detailed [here](https://en.wikipedia.org/wiki/Kernel_method#Mathematics:_the_kernel_trick). I have also modified so only a single mapping can be passed - I don't think it makes sense to map `y` and `y_pred` differently within the MMD calculation.